src/main/java/de/tum/in/www1/artemis/service/metis/conversation/ChannelService.java

```diff
@@ -0,0 +1,224 @@
+     */
+    public Channel updateChannel(Long channelId, Long courseId, ChannelDTO channelDTO) {
+        var channel = getChannelOrThrow(channelId);
+        if (channelDTO.getName() != null && !channelDTO.getName().equals(channel.getName())) {
+            channel.setName(channelDTO.getName().trim().isBlank() ? "" : channelDTO.getName().trim());
+        }
+        if (channelDTO.getDescription() != null && !channelDTO.getDescription().equals(channel.getDescription())) {
+            channel.setDescription(channelDTO.getDescription().trim().isBlank() ? "" : channelDTO.getDescription().trim());
+        }
+        if (channelDTO.getTopic() != null && !channelDTO.getTopic().equals(channel.getTopic())) {
+            channel.setTopic(channelDTO.getTopic().trim().isBlank() ? "" : channelDTO.getTopic().trim());
+        }
+        this.channelIsValidOrThrow(courseId, channel);
+
+        var updatedChannel = channelRepository.save(channel);
+        conversationService.notifyAllConversationMembersAboutUpdate(updatedChannel);
+        return updatedChannel;
+    }
+
+    /**
+     * Creates a new channel for the given course
+     *
+     * @param course  the course to create the channel for
+     * @param channel the channel to create
+     * @return the created channel
+     */
+    public Channel createChannel(Course course, Channel channel) {
+        if (StringUtils.hasText(channel.getName())) {
+            channel.setName(StringUtils.trimAllWhitespace(channel.getName().toLowerCase()));
+        }
+        final User user = this.userRepository.getUserWithGroupsAndAuthorities();
+        channel.setCreator(user);
+        channel.setCourse(course);
+        channel.setIsArchived(false);
+        this.channelIsValidOrThrow(course.getId(), channel);
+        var savedChannel = channelRepository.save(channel);
+        var conversationParticipantOfRequestingUser = new ConversationParticipant();
+        conversationParticipantOfRequestingUser.setUser(user);
+        conversationParticipantOfRequestingUser.setConversation(savedChannel);
+        conversationParticipantOfRequestingUser.setIsAdmin(true); // creator is of course admin. Special case, because creator is the only admin that can not be removed
+        conversationParticipantOfRequestingUser = conversationParticipantRepository.save(conversationParticipantOfRequestingUser);
+        savedChannel.getConversationParticipants().add(conversationParticipantOfRequestingUser);
+        savedChannel = channelRepository.save(savedChannel);
+        conversationService.broadcastOnConversationMembershipChannel(course, MetisCrudAction.CREATE, savedChannel, Set.of(user));
+        return savedChannel;
+    }
+
+    /**
+     * Checks if the given channel is valid for the given course or throws an exception
+     *
+     * @param courseId the id of the course
+     * @param channel  the channel to check
+     */
+    public void channelIsValidOrThrow(Long courseId, Channel channel) {
```

Comment:
The parameter annotation `@Valid` could simply work as well?

Reply:
Good idea! Done in: f4b71e3d764609ac40b8fdfd1ad3f67f1eebcdf5